### PR TITLE
Add OSX support

### PR DESCRIPTION
**Pull request by jdekonin**
Works toward eclipse/openj9/36

Signed-off-by: Joe Dekoning <joe_dekoning@ca.ibm.com>

### CODE REVIEW

**Comment by groeges** [comment]
@jdekonin Had a look at this PR and dont see anything that would cause issue with the build changes I ma making to get the build working on AIX (issue #61).

**Comment by jdekonin** [comment]
@jenkins test sanity plinux

**Comment by jdekonin** [comment]
@jenkins test sanity plinux

**Comment by jdekonin** [comment]
I don't see why these config changes should not be delivered.  The only question I have is with the OPENJ9_PLATFORM_CODE which I am not entirely sure what the definition should be since this is a new platform/os coverage.  The existing pattern is <os><arch><bits>, which is why I used oa64 (osx-amd-64).

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -138,6 +138,9 @@ AC_DEFUN_ONCE([OPENJ9_PLATFORM_SETUP],
     elif test "x$OPENJDK_BUILD_OS" = xwindows; then
       OPENJ9_PLATFORM_CODE=wa64
       OPENJ9_BUILDSPEC="win_x86-64_cmprssptrs"
+    elif test "x$OPENJDK_BUILD_OS" = xmacosx; then
+      OPENJ9_PLATFORM_CODE=oa64
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: Is there a related pull-request that makes `oa64` meaningful?

Comment (jdekonin) [reply closed/autoconf/custom-hook.m4]: Its used with OpenJ9.gmk '-tag:define "PLATFORM-$(OPENJ9_PLATFORM_CODE)' as an option to jpp.

Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: Yes, I knew that. Is https://github.com/eclipse/openj9/pull/262 how 'oa64' becomes meaningful?

Comment (jdekonin) [reply closed/autoconf/custom-hook.m4]: My misunderstanding, yes combined with eclipse/openj9#262, and further future changes this platform will generate a valid set of binaries.  The changes are incremental and not meant to be a final solution.

Comment (keithc-ca) [reply closed/autoconf/custom-hook.m4]: I guess the important point, though, is that this won't work without eclipse/openj9#262.

**Comment by jdekonin** [comment]
Sorry, I somehow I merged this although I don't know how.

**Pull request merged**